samples/nrf5340/empty_app_core/src/main.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <init.h>
+#include <nrf.h>
+#include <nrfx.h>
+
+/** @brief Power OFF entire RAM and suspend CPU forever.
+ *
+ * Function operates only on `register` variables, so GCC will not use
+ * RAM after function prologue. This is also true even with optimizations
+ * disabled. Interrupts are disabled to make sure that they will never
+ * be executed when RAM is powered off.
+ *
+ * @param reg_begin  Address to `POWER` register of the first NRF_VMC->RAM item
+ * @param reg_last   Address to `POWER` register of the last NRF_VMC->RAM item
+ */
+void disable_ram_and_wfi(register volatile u32_t *reg_begin,
+			 register volatile u32_t *reg_last)
+{
+	__disable_irq();
+
+	do {
+		*reg_begin = 0;
+		reg_begin += sizeof(NRF_VMC->RAM[0]) / sizeof(reg_begin[0]);
+	} while (reg_begin <= reg_last);
+
+	do {
+		__WFI();
+	} while (1);
+}
+
+void main(void)
+{
+	/* Power off RAM and suspend CPU */
+	disable_ram_and_wfi(&NRF_VMC->RAM[0].POWER,
+			    &NRF_VMC->RAM[ARRAY_SIZE(NRF_VMC->RAM) - 1].POWER);
+}
+
+#if !defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
```

Comment:
Is it necessary? You don't support ```nrf5340_dk_nrf5340_cpuappns``` target in the project CmakeList:
```
set(NRF_SUPPORTED_BOARDS
  nrf5340_dk_nrf5340_cpuapp
  )
```

Reply:
`nrf5340_dk_nrf5340_cpuappns` is possible, but there are following problem with Non-Secure firmware image:
 * It has limited access to registers and the network is stated in `spm_subimage`, so it is not possible to grant access to GPIO for the network core,
 * It is not possible to power off RAM,

General behavior will be different in NS, e.g. flashing blinky on the network core will not give you a blinking LED. I don't see that disabling application core from NS firmware is something that is really needed. That's why I decided to support only `nrf5340_dk_nrf5340_cpuapp`.

Reply:
I misunderstand you. I will delete `#if`.

Reply:
Done